sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java

```diff
@@ -269,4 +284,36 @@ public String getDataVersion() {
         return this.event;
     }
 
+    private static class JacksonSerializer implements JsonSerializer {
+        private final SerializerAdapter jacksonAdapter = JacksonAdapter.createDefaultSerializerAdapter();
+
+        @Override
+        public <T> T deserialize(InputStream stream, TypeReference<T> typeReference) {
+            try {
+                return jacksonAdapter.deserialize(stream, typeReference.getJavaType(), SerializerEncoding.JSON);
+            } catch (IOException e) {
+                throw LOGGER.logExceptionAsError(new UncheckedIOException(e));
+            }
+        }
+
+        @Override
+        public <T> Mono<T> deserializeAsync(InputStream stream, TypeReference<T> typeReference) {
+            return Mono.defer(() -> Mono.fromCallable(() -> deserialize(stream, typeReference)));
+        }
+
+        @Override
+        public void serialize(OutputStream stream, Object value) {
+            try {
+                jacksonAdapter.serialize(value, SerializerEncoding.JSON, stream);
+            } catch (IOException e) {
+                throw LOGGER.logExceptionAsError(new UncheckedIOException(e));
+            }
+        }
+
+        @Override
+        public Mono<Void> serializeAsync(OutputStream stream, Object value) {
+            return Mono.fromRunnable(() -> serialize(stream, value));
+        }
+    }
```

Comment:
This is also in azure-core. For the next release, we should avoid this duplication.

Reply:
Yes, I'll create a follow-up issue to have a default JsonSerializer in `azure-core`.